test/e2e/specs/site-editor/font-library.spec.js

```diff
@@ -86,4 +86,161 @@ test.describe( 'Font Library', () => {
+				page.getByRole( 'tab', { name: /upload/i } )
+			).toBeVisible( { timeout: 40000 } );
+
+			// Ensure font does not exist before uploading
+			if (
+				await page
+					.getByRole( 'button', { name: /commissioner/i } )
+					.isVisible( { timeout: 40000 } )
+			) {
+				await page
+					.getByRole( 'button', { name: /commissioner/i } )
+					.click();
+				await page.getByRole( 'button', { name: /delete/i } ).click();
+				await page.getByRole( 'button', { name: /delete/i } ).click();
+			}
+
+			// Upload a local font
+			await page.getByRole( 'tab', { name: /upload/i } ).click();
+			const fileChooserPromise = page.waitForEvent( 'filechooser' );
+			await page.getByRole( 'button', { name: /upload font/i } ).click();
+			const fileChooser = await fileChooserPromise;
+			await fileChooser.setFiles(
+				'./test/e2e/assets/Commissioner-Regular.ttf'
+			);
+
+			// Check font was installed successfully
+			await expect(
+				page
+					.getByLabel( 'Upload' )
+					.getByText( 'Fonts were installed successfully.' )
+			).toBeVisible( { timeout: 40000 } );
+			await page.getByRole( 'tab', { name: /library/i } ).click();
+			await expect(
+				page.getByRole( 'button', { name: /commissioner/i } )
+			).toBeVisible();
+		} );
+
+		test( 'should display the default collections tab', async ( {
+			page,
+		} ) => {
+			await expect(
+				page.getByRole( 'tab', { name: /install fonts/i } )
+			).toBeVisible( { timeout: 40000 } );
+		} );
+
+		test( 'should display the "Delete" button and delete the font', async ( {
+			page,
+		} ) => {
+			await page
+				.getByRole( 'button', { name: /commissioner/i } )
+				.click( { timeout: 40000 } );
+
+			await expect(
+				page.getByRole( 'button', { name: /delete/i } )
+			).toBeVisible( { timeout: 40000 } );
+
+			// Delete the font
+			await page.getByRole( 'button', { name: /delete/i } ).click();
+			await page.getByRole( 'button', { name: /delete/i } ).click();
+		} );
+	} );
+
+	test.describe( 'When user does not have permission to edit font families', () => {
+		test.beforeAll( async ( { requestUtils } ) => {
+			await requestUtils.activateTheme( 'twentytwentythree' );
+			await requestUtils.activatePlugin(
+				'gutenberg-test-font-library-permissions'
+			);
+		} );
+
+		test.beforeEach( async ( { admin, editor, page } ) => {
+			await admin.visitSiteEditor();
+			await editor.canvas.locator( 'body' ).click();
+			await page.getByRole( 'button', { name: /styles/i } ).click();
+			await page
+				.getByRole( 'button', { name: /typography styles/i } )
+				.click();
+			await page
+				.getByRole( 'button', {
+					name: /manage fonts/i,
+				} )
+				.click();
+		} );
+
+		test.afterAll( async ( { requestUtils } ) => {
+			await requestUtils.deactivatePlugin(
+				'gutenberg-test-font-library-permissions'
+			);
+		} );
+
+		test( 'should not display the "Upload" tab', async ( { page } ) => {
+			await expect(
+				page.getByRole( 'tab', { name: /upload/i } )
+			).toBeHidden( { timeout: 40000 } );
```

Comment:
We should probably comment to explain the need for the timeout.

Reply:
Added a couple of comments here: https://github.com/WordPress/gutenberg/pull/59316/commits/8e3f7bfb358dc75249d4ec66a126afaa3bcca31b